docs/upgrade-to-v4.md

```diff
@@ -0,0 +1,185 @@
+# Microsoft Graph .NET SDK v4 changelog and upgrade guide
+
+The purpose of this document is to outline any breaking change and migration work SDK users might run into while upgrading to the v4 of the SDK.
+
+## Breaking changes
+
+To improve the development experience provided by the SDK, it was necessary to make the following breaking changes in addition to the changes listed in the migration guide below:
+
+ * .NET Standard minimum version bumped from `netstandard1.3` to `netstandard2.0`
+ * .NET Framework minimum version bumped from `net45` to `net461`
+ * Replacing Newtonsoft.Json dependency with System.Text.Json for serialization/de-serialization
+ * Upgrading Microsoft.Graph.Core dependency to version 2.0.0
+
+## Upgrade guide for breaking changes
+
+The following section lists out the breaking changes requiring code changes from SDK users.
+
+### System.Text.Json replaces Newtonsoft.Json
+
+Updating to the latest version of the library involves the migration from [Newtonsoft.Json](https://www.newtonsoft.com/json) to System.Text.Json. You can always read more about the differences between the two libraries [here](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to?pivots=dotnet-5-0#table-of-differences-between-newtonsoftjson-and-systemtextjson).
+
+From the experience of the user of the SDKs, these are the main changes to look out for when upgrading.
+
+1. Object types and function parameters using Newtonsoft's JToken are now replaced by System.Text.Json's JsonDocument.
+
+For example, the code sample :-
+
+```cs
```

Comment:
if you want those code blocks to be "under" the bullet point, you need to indent with 2 spaces. Same thing for the "For example" and "would change to" texts

Reply:
Thanks! Resolved via https://github.com/microsoftgraph/msgraph-sdk-dotnet/pull/890/commits/38e8635bae1b231b3659d52a34c9662d8b282538